src/nimble.nim

```diff
@@ -547,12 +547,22 @@ proc search(options: Options) =
   if needsRefresh(options):
     raise newException(NimbleError, "Please run nimble refresh.")
   let pkgList = getPackageList(options)
-  var found = false
+  var
+    found = false
+    jsonOutput: JsonNode
+  if options.jsonOutput:
+    jsonOutput = newJArray()
   template onFound {.dirty.} =
-    echoPackage(pkg)
-    if pkg.alias.len == 0 and options.queryVersions:
-      echoPackageVersions(pkg)
-    echo(" ")
+    if options.jsonOutput:
+      let info = pkg.json
+      if pkg.alias.len == 0 and options.queryVersions:
+        info["versions"] = packageVersionsJson(pkg)
```

Comment:
Why not do this in the above `json` function?

Reply:
There is a problem with order of includes, and I prefer declaring `json` as a `func`.

Reply:
Order of includes? Not sure what you mean.

Reply:
There would be an import loop unless I move `toJson` to `download.nim`.